internal/runbits/projectmigration/projectmigration.go

```diff
@@ -0,0 +1,68 @@
+package projectmigration
+
+import (
+	"path/filepath"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/multilog"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/pkg/localcommit"
+	"github.com/ActiveState/cli/pkg/projectfile"
+)
+
+type projecter interface {
+	URL() string
+	Path() string
+	LegacyCommitID() string
+}
+
+type promptable interface {
+	Confirm(title, message string, defaultChoice *bool) (bool, error)
+}
+
+// PromptToMigrateIfNecessary checks if the given project has been migrated to the new project
+// format. If not, the user is prompted to do so. This should be called in main() as soon as a
+// project file is parsed.
+func PromptToMigrateIfNecessary(proj projecter, prompt promptable, out output.Outputer) error {
+	projectDir := filepath.Dir(proj.Path())
+	if _, err := localcommit.Get(projectDir); err == nil || !localcommit.IsFileDoesNotExistError(err) {
+		return err
+	}
+
+	defaultChoice := false
+	if migrate, err := prompt.Confirm("", locale.T("projectmigration_confirm"), &defaultChoice); err == nil && !migrate {
+		if out.Config().Interactive {
+			out.Notice(locale.Tl("projectmigration_declined", "Migration declined for now"))
+		}
+		return nil
+	} else if err != nil {
+		return locale.WrapError(err, "err_projectmigration_confirm", "Could not confirm migration choice")
```

Comment:
We shouldn't be locale wrapping anymore, unless it's a vague error but we think it contains valuable information. This can probably just be an errs.Wrap().